### PR TITLE
Feature - Create award category test

### PR DESCRIPTION
**Pull request by at-anhquach**

### CODE REVIEW

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import vn.asiantech.base.DriverBase;
+import vn.asiantech.page.AwardCategoryPage;
+
+public class AwardCategoryDefinitions extends DriverBase implements En {
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L13"></span>Missing a Javadoc comment.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        Then("^Submit change from disable to enable$", () -> Assert.assertTrue(awardCategoryPage.isEnableSubmitButton()));
+
+        Given("^New dialog is showed and entered name$", () -> {
+            awardCategoryPage.clickNewAward();
+            awardCategoryPage.enterName("Best project" + System.currentTimeMillis());
+        });
+
+        When("^I click submit button$", () -> awardCategoryPage.clickSubmit());
+
+        Then("^Table add one line$", () -> Assert.assertTrue(awardCategoryPage.checkAddLine(countCurrentLine+1)));
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L53"></span>’+’ is not preceded with whitespace.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L53"></span>’+’ is not followed by whitespace.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        When("^I clear name box$", () -> awardCategoryPage.clearText("name"));
+
+        Then("^Display message \"([^\"]*)\"$", (String message) -> Assert.assertTrue(awardCategoryPage.isShowError()));
+
+        And("^Disable submit button$", () -> Assert.assertFalse(awardCategoryPage.isEnableSubmitButton()));
+
+        But("^I click close button$", () -> awardCategoryPage.clickCloseDialog());
+
+        Then("^No line is added to table$", () -> Assert.assertTrue(awardCategoryPage.checkAddLine(countCurrentLine+1)));
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L71"></span>’+’ is not preceded with whitespace.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L71"></span>’+’ is not followed by whitespace.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    private void displayPage() {
+        driver.get("http://portal-stg.asiantech.vn" + "/admin/award-category");
+        new WebDriverWait(driver, TIMEOUT_IN_SECONDS).until(
+                webDriver -> ((JavascriptExecutor) webDriver).executeScript("return document.readyState").equals("complete"));
+        new WebDriverWait(driver, TIMEOUT_IN_SECONDS).until(ExpectedConditions.visibilityOfElementLocated(By.id("page-wrapper")));
+        Assert.assertEquals("http://portal-stg.asiantech.vn" + "/admin/award-category", driver.getCurrentUrl());
+    }
+
+    private void setCountCurrentLine(int count) {
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/stepdefs/AwardCategoryDefinitions.java#L88"></span>Parameter count should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/AwardCategoryPage.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import vn.asiantech.base.BasePage;
+
+import java.util.List;
+
+public class AwardCategoryPage extends BasePage<AwardCategoryPage> {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/AwardCategoryPage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/fa98d01706a4d73b6924e521d26d334d791b9029/src/test/java/vn/asiantech/page/AwardCategoryPage.java#L12"></span>Missing a Javadoc comment.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/61876ba2d800ece60cd13413dc9048748b6c4307/src/test/java/stepdefs/AwardCategoryDefinitions.java#L93"></span>Parameter count should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/AwardCategoryPage.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    public final AwardCategoryPage navigateTo(final WebDriver webDriver) {
+        return null;
+    }
+
+    public final int getCountLine() {
+        return (tbCategory.findElements(By.tagName("tr")).size());
+    }
+
+    public final boolean checkAddLine(final int count) {
+        return getCountLine() < count+1;
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/AwardCategoryPage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/61876ba2d800ece60cd13413dc9048748b6c4307/src/test/java/vn/asiantech/page/AwardCategoryPage.java#L56"></span>’+’ is not preceded with whitespace.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/AwardCategoryPage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/61876ba2d800ece60cd13413dc9048748b6c4307/src/test/java/vn/asiantech/page/AwardCategoryPage.java#L56"></span>’+’ is not followed by whitespace.</td>
    </tr>
  </tbody>
</table>

**Comment by at-hanhnguyen2** [comment]
@at-hangtran : please check again!

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+import vn.asiantech.base.DriverBase;
+import vn.asiantech.page.AwardCategoryPage;
+
+/**
+ * @author at-anh.quach
+ * AwardCategoryDefinitions
+ */
+
+public class AwardCategoryDefinitions extends DriverBase implements En {
+    private static final int TIMEOUT_IN_SECONDS = 10;
```
Comment (at-hangtran) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: Use `DEFAULT_TIME_OUT` in Constant class

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    private AwardCategoryPage awardCategoryPage;
+    private WebDriver driver;
+
+    public AwardCategoryDefinitions() {
+        try {
+            driver = getDriver();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        awardCategoryPage = initPage(getDriver(), AwardCategoryPage.class);
```
Comment (at-hangtran) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: use `driver`

File: src/test/java/stepdefs/AwardCategoryDefinitions.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        Then("^Alert category has been created show$", () -> Assert.assertTrue(awardCategoryPage.isDisplayAlertSuccess()));
+
+        When("^I input name was exist on Name box$", () -> awardCategoryPage.enterName("Best project"));
+
+        Then("^Alert danger show$", () -> Assert.assertTrue(awardCategoryPage.isDisplayAlertDanger()));
+    }
+
+    private void displayPage() {
+        driver.get(Constant.AWARD_CATEGORY_PAGE_URL);
+        new WebDriverWait(driver, TIMEOUT_IN_SECONDS).until(
```
Comment (at-hangtran) [inline src/test/java/stepdefs/AwardCategoryDefinitions.java]: Use `waitForPageDisplayed`  in DriverBaseDriverBase class

File: src/test/java/vn/asiantech/page/AwardCategoryPage.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    @FindBy(id = "award-list-wrapper")
+    private WebElement tbCategory;
+
+    @FindBy(id = "btn-create-award")
+    private WebElement btnNewAward;
+
+    @FindBy(id = "btn-submit-award")
+    private WebElement btnSubmit;
+
+    @FindBy(css = ".btn.btn-sm.btn-default.btn-cancel")
```
Comment (at-hangtran) [inline src/test/java/vn/asiantech/page/AwardCategoryPage.java]: Use `@FindBy(className = "btn-cancel")` is enough

File: src/test/java/vn/asiantech/page/AwardCategoryPage.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+    private WebElement btnCancel;
+
+    @FindBy(name = "name")
+    private WebElement inputName;
+
+    @FindBy(css = ".form-control.ng-untouched.ng-pristine.ng-valid")
```
Comment (at-hangtran) [inline src/test/java/vn/asiantech/page/AwardCategoryPage.java]: Use `@FindBy(className = "ng-valid")` is enough

**Pull request merged**